src/sagas/projectsSaga.js

```diff
@@ -0,0 +1,68 @@
+import { call, put, takeLatest, all } from 'redux-saga/effects';
+import { getAllProjects } from '../service/projectsService';
+import {
+  actionTypes
+} from '../components/projects/modules/types';
+import {
+  fetchProjectsSuccess,
+  fetchProjectsFailure,
+  fetchProjectsRequest, 
+} from '../components/projects/modules/actions';
+
+
+function* fetchProjectsSaga(action) {
+  const { page, pageSize, selectedProjectStatus, searchInput } = action.payload;
```

Comment:
without deconstructing this, you could pass an object to service and with that you can reduce number of parameters because best function is one without parameters